components/patient-access-rules/api/src/test/java/org/phenotips/data/permissions/internal/DefaultCollaboratorTest.java

```diff
@@ -167,6 +167,32 @@ public void equalsTest() throws ComponentLookupException
         Assert.assertFalse(c.equals("other"));
     }
 
+    /** Basic tests for {@link Collaborator#equals(Object)}. */
+    @Test
+    public void equalsWithNullUserTest() throws ComponentLookupException
+    {
+        Collaborator c = new DefaultCollaborator(null, this.access, this.helper);
+        // Equals itself
+        Assert.assertTrue(c.equals(c));
+        // Doesn't equal null
+        Assert.assertFalse(c.equals(null));
+        // Equals an identical collaborator
+        AccessLevel otherAccess = mock(AccessLevel.class);
+        Collaborator other = new DefaultCollaborator(null, this.access, this.helper);
+        Assert.assertTrue(c.equals(other));
+        // Doesn't equal a collaborator with same user but different access
+        other = new DefaultCollaborator(null, otherAccess, this.helper);
+        Assert.assertFalse(c.equals(other));
+        // Doesn't equal a collaborator with same access but different user
+        other = new DefaultCollaborator(new DocumentReference("xwiki", "XWiki", "padams"), this.access, this.helper);
```

Comment:
I'd like to see a test in the other direction as well, either `other equals c` here, or add `c equals new collaborator(null, access, helper)` above in `equalsTest`.

Reply:
Fixed.